### PR TITLE
Add typing check

### PR DESCRIPTION
**Pull request by PhilippeMoussalli**
This PR adds type checking into the CICD pipelines of Fondant. We're leveraging the latest version of mypy. 
#46

### CODE REVIEW

File: fondant/component_spec.py [hunk 1]
```diff
@@ -239,7 +244,7 @@ def from_fondant_component_spec(
     @staticmethod
     def _dump_args(args: t.List[Argument]) -> t.List[t.Union[str, t.Dict[str, str]]]:
         """Dump Fondant specification arguments to kfp command arguments."""
-        dumped_args = []
+        dumped_args: t.List[t.Union[str, t.Dict[str, str]]] = []
```
Comment (NielsRogge) [inline fondant/component_spec.py:247]: Tbh I find these typing hints incredibly hard to read and I'm not sure they're helpful for us. They seem useful for computers but not for humans
 
`t.List[t.Union[str, t.Dict[str, str]]]` => I mean just look at this

Just my personal opinion :D

Comment (PhilippeMoussalli) [reply fondant/component_spec.py:247]: I twas getting errors from `mypy` as follows 
 Incompatible return value type (got "List[str]", expected "List[Union[str, Dict[str, str]]]") which went away when I explicitly defined the type.  
However, the args of kubeflow does required to have a list that can contains either a string or a dictionary so in that sense I think the type hint in the function's return is valid.

I'll just add a mypy exception for that one

**Review by RobbeSneyders — APPROVED** [review]
Thanks @PhilippeMoussalli!

I would revert ignoring the args type, but otherwise LGTM.

File: fondant/component_spec.py (hunk 1 above)
Comment (PhilippeMoussalli) [reply fondant/component_spec.py]: alright I think that's a better approach indeed :) thanks for the suggestion. Added

**Pull request merged**